### PR TITLE
Use async def in pythoneval tests

### PR DESCRIPTION
**Pull request by hauntsaninja**
Helps with Python 3.11 support

### CODE REVIEW

File: test-data/unit/pythoneval-asyncio.test [hunk 1]
```diff
@@ -17,8 +17,7 @@ from typing import Any, Generator
 import asyncio
 from asyncio import Future
 
-@asyncio.coroutine
-def greet_every_two_seconds() -> 'Generator[Any, None, None]':
+async def greet_every_two_seconds() -> 'Generator[Any, None, None]':
```
Comment (JelleZijlstra) [inline test-data/unit/pythoneval-asyncio.test:20]: Can async defs still do `yield from`? I think you need `await` below.

Comment (hauntsaninja) [reply test-data/unit/pythoneval-asyncio.test:20]: They cannot. Sorry, I should have marked this PR as a draft. I just did a find and replace and local tests have been finnicky and slow for a long time

**Comment by hauntsaninja** [comment]
Okay, this is ready for review. I filed https://github.com/python/mypy/issues/12837 for the one xfailed test. It's easy to fix (if we want to), but that should be its own discussion and PR.

Note that this leaves all asyncio.coroutine behaviour untested, which I assume is fine. I also haven't checked to see if there's testing overlap with `check-async-await.test`

File: test-data/unit/pythoneval-asyncio.test [hunk 2]
```diff
@@ -4,7 +4,7 @@
 -- These are mostly regression tests -- no attempt is made to make these
 -- complete.
 --
--- This test file check Asyncio and yield from interaction
+-- This test file check Asyncio and await interaction
```
Comment (AlexWaygood) [inline test-data/unit/pythoneval-asyncio.test:7]:
```suggestion
-- This test file checks Asyncio and await interaction
```

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**